src/com/squareup/ideaplugin/dagger/handler/ConstructorInjectToProvidesHandler.java

```diff
@@ -45,8 +42,36 @@
   }
 
   private void showUsages(MouseEvent mouseEvent, PsiParameter psiParameter) {
-    new ShowUsagesAction(new Decider.ConstructorParameterInjectDecider(psiParameter)).startFindUsages(
-        PsiConsultantImpl.checkForLazyOrProvider(psiParameter), new RelativePoint(mouseEvent),
-        PsiUtilBase.findEditor(psiParameter), MAX_USAGES);
+    // Check to see if class type of psiParameter has constructor with @Inject. Otherwise, proceed.
+    if (!navigateToConstructorIfProvider(psiParameter)) {
+      new ShowUsagesAction(new Decider.ConstructorParameterInjectDecider(psiParameter)).startFindUsages(
+              PsiConsultantImpl.checkForLazyOrProvider(psiParameter), new RelativePoint(mouseEvent),
+              PsiUtilBase.findEditor(psiParameter), MAX_USAGES);
+    }
+  }
+
+  private boolean navigateToConstructorIfProvider(PsiParameter psiParameter) {
+    PsiTypeElement declaringTypeElement = psiParameter.getTypeElement();
+    PsiClass classElement = JavaPsiFacade.getInstance(psiParameter.getProject()).findClass(
```

Comment:
Just tried it with psiParameter and couldn't get it to work. I tried some other various methods too and could not get them to work.